feedTheMonster.ts

```diff
@@ -30,22 +36,29 @@ class App {
     this.versionInfoElement = document.getElementById("version-info-id") as HTMLElement;
     this.loadingElement = document.getElementById('loading-screen') as HTMLElement;
     this.is_cached = this.initializeCachedData();
+    this.firebaseIntegration = new FirebaseIntegration();
+    this.startSessionTime = 0;
+    this.init();
     this.channel.addEventListener("message", this.handleServiceWorkerMessage);
     window.addEventListener("beforeunload", this.handleBeforeUnload);
-    this.init();
+    document.addEventListener("visibilitychange", this.handleVisibilityChange);
```

Comment:
Can we use VISIBILITY_CHANGE from "../common/event-names" instead of using "visible" ?